### PR TITLE
add: mmi playwright e2e tests

### PR DESCRIPTION
**Pull request by racitores**
## **Description**
This PR adds playwright test capabilities for the mmi extension build. It also adds functional and visual tests for that build. This is part of the MMI e2e legacy tests.

NOTE: The new test set will be added to the current pipeline as a non-blocking step. This means that even if these tests fail, other Pull Requests can still be merged. However, a failure will trigger a warning indicating potential issues with MMI functionality due to the changes in these PRs.

For more detailed info about how the MMI e2e tests works, check the README file within the `test/e2e/mmi` folder. 

## **Manual testing steps**

N/A

## **Screenshots/Recordings**

N/A
### **Before**

N/A
### **After**

N/A
## **Related issues**

N/A
## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by racitores** [comment]
@metamaskbot update-policies

**Comment by racitores** [comment]
@metamaskbot update-policies

**Comment by racitores** [comment]
@SocketSecurity ignore-all

File: .gitignore
Comment (zone-live) [inline .gitignore:73]: this is repeated 🤔

File: test/e2e/mmi/helpers/oauth-api-helper.ts
Comment (zone-live) [inline test/e2e/mmi/helpers/oauth-api-helper.ts:14]: From where does the env variable value comes from?

**Review by zone-live — DISMISSED** [review]

File: test/e2e/mmi/helpers/oauth-api-helper.ts
Comment (racitores) [reply test/e2e/mmi/helpers/oauth-api-helper.ts:14]: Locally a .env file is required. In the pipeline all env vars come from CircleCI environment variables setup.

Comment (zone-live) [reply test/e2e/mmi/helpers/oauth-api-helper.ts:14]: Ah got it! 👍🏼

**Comment by racitores** [comment]
@metamaskbot update-policies

**Review by zone-live — DISMISSED** [review]

File: package.json
Comment (Gudahtt) [inline package.json:111]: Merge mistake?

Comment (racitores) [reply package.json:111]: No, it is intentional. We have updated the Playwright version and, in order to make it compatible with the Storybook config, I had to add this command in the CI.

File: .circleci/config.yml
Comment (racitores) [reply .circleci/config.yml:637]: As far as I could see, the test build is by default trying to connect to local testnet and if I try to change to any other network it fails. This is one of the main reasons. I also think that testing the prod-like version gives more value that a test one. But I am happy to consider other approach if is better or more aligned with the builds types rational.

File: package.json
Comment (Gudahtt) [reply package.json:111]: This command doesn't appear to be related to Playwright or Storybook though

Comment (racitores) [reply package.json:111]: ah, sorry, you are completely right! I'll remove the line

**Review by zone-live — DISMISSED** [review]

File: test/e2e/mmi/README.md
Comment (PeterYinusa) [inline test/e2e/mmi/README.md:22]: To run the tests, I first had to run `yarn playwright install` to download Chromium.

File: package.json
Comment (PeterYinusa) [inline package.json:113]: NIT: Would be easier to find if it was grouped with the other `test:*` scripts

File: package.json
Comment (PeterYinusa) [inline package.json:471]: Question: Could we have leveraged `.metamaskrc` to load environment variables instead of introducing `dotenv`?

Comment (racitores) [reply package.json:471]: I believe the .metamaskrc file is designated for MetaMask-related configurations, whereas the .env file is utilised for end-to-end Playwright testing. My preference is to maintain these configurations separately for clarity and organisation.

File: package.json
Comment (racitores) [reply package.json:48]: this is update with the latest rebase 👍 
BTW, the documentation (README) is outdated related this topic

**Review by zone-live — DISMISSED** [review]

File: Dockerfile
Comment (brad-decker) [inline Dockerfile:1]: How is this Dockerfile used?

**Comment by brad-decker** [comment]
I am adding the DO NOT MERGE label here just so people can feel free to approve PR if they are in agreement without signaling a merge while there are ongoing conversations happening. I'll remove as soon as we have alignment.

File: .circleci/config.yml
Comment (brad-decker) [inline .circleci/config.yml:30]: I'm assuming this is to announce in a slack channel about test status?

File: .circleci/config.yml
Comment (brad-decker) [inline .circleci/config.yml:1093]: Develop tests run **very frequently**... are we sure this won't become noise v signal?

File: .circleci/config.yml
Comment (brad-decker) [inline .circleci/config.yml:1081]: This is fine for now, but would be cool to see if we can somehow use the circleci commands to split tests so we can benefit from the "rerun failed tests" command that only reruns failed tests versus failed jobs. Not a blocker for this PR

**Review by brad-decker — CHANGES_REQUESTED** [review]

**Comment by brad-decker** [comment]
Removing DO NOT MERGE flag as we agreed in slack to figure out the best way to land this and reach alignment shortly there after. @racitores if possible could you update the PR description to indicate that these tests are considered legacy tests that we need to run against MMI builds until we reach alignment? IT may very well believe we align more of the codebase towards playwright but until we reach that decision I want to make it crystal clear in the documentation that we aren't intending on adopting this pattern (yet)

File: .circleci/config.yml
Comment (racitores) [reply .circleci/config.yml:30]: yes, correct

File: .circleci/config.yml
Comment (racitores) [reply .circleci/config.yml:1093]: As the job is optional, we need to know when the mmi build has started failing on develop (to identify cases where non mmi changes impacts the mmi build). Therefore we need this notification. Notice that will be only notified in case of failure of this job, so hopefully won't be so frequent, and MMI team should make sure it is not failing for long time (so mmi team should fix the issue asap)

File: .gitignore
Comment (racitores) [reply .gitignore:73]: Updated. Could have been lost due to a rebase. Good catch.

File: .circleci/config.yml
Comment (racitores) [reply .circleci/config.yml:1081]: I didn't know that feature, nice! I'll create a ticket on my backlog to add it after the PR is merged

File: test/e2e/mmi/package.json
Comment (racitores) [reply test/e2e/mmi/package.json:2]: I have updated the code so now tests dependencies are within the dockerfile image. Beside that I have added a little note in the main README that points to the specific README of the MMI e2e tests.

File: .gitignore
Comment (racitores) [reply .gitignore]: Updated

**Review by brad-decker — APPROVED** [review]
With the additional changes I think this looks good for merge and is minimally intrusive to the current code structure. Lets continue conversations async about how we can move more of MetaMask tests to use Playwright (if its viable) and align on a single test pattern framework for E2E

**Review by zone-live — APPROVED** [review]

File: package.json
Comment (PeterYinusa) [reply package.json:472]: 💯

**Pull request merged**
